src/univariate/continuous/ekdist.jl

```diff
@@ -0,0 +1,224 @@
+        
+convert(::Type{EKDist{T}}, d::EKDist) where {T<:Real} = 
+        EKDist{T}(T(d.α), T(d.β), T(d.γ), T(d.l), T(d.u))
+
+convert(::Type{EKDist{T}}, d::EKDist{T}) where {T<:Real} = d
+
+#### Parameters
+
+params(d::EKDist) = (d.α, d.β, d.γ, d.l, d.u)
+partype(::EKDist{T}) where {T} = T
+
+#### Evaluation
+
+function pdf(d::EKDist, x::Real)
+      xs = (x - d.l) / (d.u - d.l)
+      α = d.α
+      β = d.β
+      γ = d.γ
+    return ( α * β * γ * 
+             xs^(α - 1) * 
+             (1 - xs^α)^(β - 1) * 
+             (1 - (1 - xs^α)^β)^(γ - 1)
+           )
+end
+
+function logpdf(d::EKDist, x::Real)
+      xs =  (x - d.l) / (d.u - d.l) 
+      α = d.α
+      β = d.β
+      γ = d.γ
+    return  ( log(α) + log(β) + log(γ) + 
+              log(xs)*(α - 1) + 
+              log(1 - xs^α)*(β - 1) + 
+              log(1 - (1 - xs^α)^β)*(γ - 1)
+            ) 
+end
+
+function logpdf(d::EKDist, x::Real, edgebound::Real)
+    # Failes at exactly 1 or 0, so I guess jitter it by small amount
+    # this only matters to me in likelihood analysis for real data
+    # that might get defined as the max of the data or Fitting
+    # bound parameters as part of the process.
+    xc = clamp( x, d.l + edgebound, d.u - edgebound)
+    return logpdf(d, xc)  
+end
+
+# do we even need gradlogpdf?
+# grad implied gradient but this is definitely derivative wrt x
+function gradlogpdf(d::EKDist, x::Real)
+    outofsupport = (x < d.l) | (x > d.u)
+    xs = (x - d.l) / (d.u - d.l)
+    α = d.α
+    β = d.β
+    γ = d.γ
+
+    z = (α - 1) / xs +
+        -(α * xs^(α - 1)) / (1 - xs^α) * (β - 1) +
+        (α * β * xs^(α - 1)) * (1 - xs^α)^(β - 1) / (1 -(1 - xs^α)^β) * (γ - 1)
+    
+    return outofsupport ? zero(z) : z
+end
+
+function gradlogpdf(d::EKDist, x::Real, edgebound::Real)
+    xc = clamp( x, d.l + edgebound, d.u - edgebound)
+    return gradlogpdf(d, xc)
+end
+
+
+
+function cdf(d::EKDist, x::Real)
+      xs = (x - d.l) / (d.u - d.l)
+      α = d.α
+      β = d.β
+      γ = d.γ
+    return (1 - ( 1 - xs^α)^β)^γ
+end
+
+function ccdf(d::EKDist, x::Real)
+      xs = (x - d.l) / (d.u - d.l)
+      α = d.α
+      β = d.β
+      γ = d.γ
+
+    return 1 - (1 - ( 1 - xs^α)^β)^γ
+end
+
+function logcdf(d::EKDist, x::Real)
+    return log(cdf(d, x))
+end
+
+function logccdf(d::EKDist, x::Real)
+    return log(ccdf(d, x))
+end
+
+function quantile(d::EKDist, q::Real)
+    α = d.α
+    β = d.β
+    γ = d.γ
+    xs = ( 1 - (1 - q^(1/γ) )^(1/β) )^(1/α) 
+    x = xs * (d.u - d.l) + d.l
+    return x
+end
+
+cquantile(d::EKDist, q::Real) = quantile(d, 1 - q)
+
+
+#### Sampling
+# `rand`: Uses fallback inversion sampling method
+
+## Fitting
+
+function fit_mle(::Type{<:EKDist}, x::AbstractArray{T}) where T<:Real
+    #incomplete
+end
+
+
+#### Statistics
+median(d::EKDist) = quantile(d, .5) # right?
+
+function ekdmoment(d::EKDist, k::Real)
+    α = d.α
+    β = d.β
+    γ = d.γ
+
+    # can't tell from paper if its infinite sum or stops at the floor of γ -1
```

Comment:
Which part of the paper are you looking at? They looked like infinite sums to me, but maybe I missed something.